scripts/sct_maths.py

```diff
@@ -472,7 +506,65 @@ def laplacian(data, sigmas):
     # from scipy.ndimage.filters import laplace
     # return laplace(data.astype(float))
 
+def compute_similarity(data1, data2, fname_out='', metric='', verbose=1):
+    assert data1.size == data2.size, "ERROR: the data don't have the same size"
+    data1_1d = data1.ravel()
+    data2_1d = data2.ravel()
+
+    if metric == 'mi':
+        res = calc_MI(data1_1d, data2_1d, normalized=True)
+        metric_full = 'Mutual information'
+    if metric == 'corr':
+        res = calc_corr(data1_1d, data2_1d)
+        metric_full = 'Pearson correlation coefficient'
+
+    printv('\n'+ metric_full +': ' + str(res), verbose, 'info')
+
+    path_out, filename_out, ext_out = extract_fname(fname_out)
+    if ext_out not in ['.txt', '.pkl', '.pklz', '.pickle']:
+        printv('ERROR: the output file should a text file or a pickle file. Received extension: '+ext_out, 1, 'error')
+
+    elif ext_out == '.txt':
+        file_out = open(fname_out, 'w')
+        file_out.write(metric_full+': \n'+str(res))
+        file_out.close()
+
+    else:
+        import pickle, gzip
+        if ext_out == '.pklz':
+            pickle.dump(res, gzip.open(fname_out, 'wb'), protocol=2)
+        else:
+            pickle.dump(res, open(fname_out, 'w'), protocol=2)
+
+def calc_MI(x, y, nbins=32, normalized=False):
```

Comment:
Same argument for correlation. Full name are better in my point of view, as they are more understandable.